src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -646,18 +649,18 @@ void Connection::initiate_reconnect()
     }
 
     m_websocket =
-        m_client.m_socket_factory.connect(this, util::websocket::EZEndpoint{
```

Comment:
maybe assert that the `m_client.m_socket_factory` is not a nullptr, either here before using it, or in the constructor

Reply:
Asserted in constructor